File: testdata/jennies/builders/envelope_assignment/PythonBuilder/builders/sandbox.py
import typing
from ..cog import builder as cogbuilder
from ..models import sandbox


class Dashboard(cogbuilder.Builder[sandbox.Dashboard]):
    _internal: sandbox.Dashboard

    def __init__(self):
        self._internal = sandbox.Dashboard()

    def build(self) -> sandbox.Dashboard:
        """
        Builds the object.
        """
        return self._internal    
    
    def with_variable(self, name: str, value: str) -> typing.Self:    
        if self._internal.variables is None:
            self._internal.variables = []
        
        self._internal.variables.append(sandbox.Variable(
            name=name,
            value=value,
        ))
    
        return self
    

File: testdata/jennies/builders/envelope_assignment/RustBuilder/src/builders/sandbox.rs
use crate::cog;
use crate::types::sandbox;

#[derive(Debug, Clone)]
pub struct DashboardBuilder {
    internal: sandbox::Dashboard,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl DashboardBuilder {
    pub fn new() -> Self {
        Self {
            internal: sandbox::Dashboard::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for DashboardBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl DashboardBuilder {
    pub fn with_variable(mut self, name: String, value: String) -> Self {
        self.internal
            .variables
            .push(sandbox::Variable { name, value });

        self
    }
}

impl cog::Builder<sandbox::Dashboard> for DashboardBuilder {
    fn build(&self) -> Result<sandbox::Dashboard, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}
